app/main/posts/modify/post-data-editor.controller.js

```diff
@@ -84,20 +86,51 @@ function PostDataEditorController(
         $scope.savePost();
     });
 
-    $scope.$on('event:edit:leave:form', function () {
-        if ($scope.parentForm.form && $scope.parentForm.form.$dirty) {
-            $scope.leavePost();
-        } else {
-            $scope.cancel();
+    $transitions.onStart({}, function (transition) {
+        //where is it going? transition.to().name
+        // return rejected promise or false to cancel the transition
+        // leavePost calls cancel which then resolves or rejects the state change.
+        if (!ignoreCancelEvent && transition.from().name === 'list.data.edit') {
+            return $scope.leavePost(transition);
         }
-    });
 
-    /**
-     * @uirouter-refactor check this and add back if neded
-     * var $locationChangeStartHandler = $scope.$on('$locationChangeStart', function (e, next) {
-        $scope.leavePost(next, e);
+        return true;
+
     });
-     */
+
+    function leavePost(transition) {
+        /**
+         * just a wrapper function because we were doing the same thing
+         * over and over again in the promise
+         * @param resolve
+         * @param reject
+         */
+        var cancelResolveFn = function (resolve, reject) {
+            $scope.isLoading.state = false;
+            $scope.savingPost.saving = false;
+            return $scope.cancel().then(function () {
+                resolve(true);
+            }).catch(function () {
+                reject(false);
+            });
+        };
+
+        return new Promise (function (resolve, reject) {
+            if (!$scope.parentForm.form || ($scope.parentForm.form && !$scope.parentForm.form.$dirty)) {
+                cancelResolveFn(resolve, reject);
+            } else {
+                // @uirouter-refactor if we end up having onbeforeunload features,we need to add this back
+                // if (ev) {
+                //     ev.preventDefault();
+                // }
+                Notify.confirmLeave('notify.post.leave_without_save').then(function () {
```

Comment:
How does this work if both confirm or cancel lead to the same function??

Reply:
for 25% of the cases, it doesn't =) 
thanks, I had missed the fact that because there's no edit mode bool I needed to do a reject .

Reply:
I'm confused. Why do both save and continue with out saving follow the same path? Where do we actually trigger saving?

Reply:
in the confirmLeave function there's a save function that resolves or reject depending on whether or not your save is successful. So resolve means it saved and you can continue. Reject means it didn't save and you CANNOT continue. 
 It also has a confirm function that always resolves to let the user continue without saving.